src/main/java/com/project/bookstore/repository/book/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,24 @@
+package com.project.bookstore.repository.book;
+
+import com.project.bookstore.model.Book;
+import com.project.bookstore.repository.SpecificationProvider;
+import com.project.bookstore.repository.SpecificationProviderManager;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException(

```

Comment:
avoid one-letter naming even in stream chains, change to `provider`
let's throw more specific noSuchElement exception